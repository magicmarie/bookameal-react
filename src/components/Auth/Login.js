import React from "react";
import { notify } from "react-notify-toast";
import jwtDecode from "jwt-decode";
import { Link } from "react-router-dom";
import { AppContext } from "../../appContext";
import axiosInstance from "../common/Apicalls";

/**
 *@param {Event} event
 * @class Login
 * @extends {React.Component}
 */
export class Login extends React.Component {
  state = {
    email: "",
    password: "",
    errors: {}
  };

  onChange = event =>
    this.setState({ [event.target.name]: event.target.value });

  // login
  handleLogin = event => {
    event.preventDefault();
    const { email, password } = this.state;
    axiosInstance
      .post("/auth/login", { email, password })
      .then(response => {
        // save token to localstorage, decode to get is_admin value
        localStorage.setItem("token", response.data.token);
        this.props.context.login(response.data.token);
        const decoded = jwtDecode(response.data.token);
        // admin/caterer
        notify.show(response.data.message, "success", 2000);
        const redirectUrl = decoded.is_admin === "True" ? "/admin" : "/user";
        this.props.history.push(redirectUrl);
      })
      .catch(error => {
        if (error.response) {
          notify.show(error.response.data.message, "error", 2500);
        }
      });
  };

  /**
   * @returns {any} rendered items
   * @memberof Login
   */
  render() {
    const { email, password } = this.state;
    return (
      <div className="row justify-content-md-center">
        <div className="col-md-5">
          <div className="jumbotron">
            <form onSubmit={this.handleLogin}>
              <h2 className="text-center form-signin-heading">
                Log in to your account
              </h2>
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input
                  type="email"
                  value={email}
                  onChange={this.onChange}
                  name="email"
                  required
                  className="form-control"
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  value={password}
                  onChange={this.onChange}
                  name="password"
                  required
                  className="form-control"
                />
              </div>
              <div>
                <button type="submit" className="btn btn-primary btn-block">
                  Login
                </button>
              </div>
              <p />
              <p className="text-center">
                Do not have an account yet?
                <Link href="/signup" to="/signup">
                  {" "}
                  Sign Up
                </Link>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default React.forwardRef((props, ref) => (
  <AppContext.Consumer>
    {context => <Login {...props} context={context} ref={ref} />}
  </AppContext.Consumer>
));
