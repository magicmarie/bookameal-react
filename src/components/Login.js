import React from "react";
import axios from "axios";
import { notify } from "react-notify-toast";
import jwtDecode from "jwt-decode";
import { Link } from "react-router-dom";
import { AppContext } from "../appContext";

class Login extends React.Component {
  state = {
    email: "",
    password: "",
    errors: {}
  };
  // login
  handleLogin = event => {
    event.preventDefault();
    const { email, password } = this.state;
    axios
      .post("/api/v1/auth/login ", { email, password })
      .then(response => {
        // save token to localstorage, decode to get is_admin value
        localStorage.setItem("token", response.data.token);
        this.props.context.login(response.data.token);
        const decoded = jwtDecode(response.data.token);
        // admin/caterer
        if (decoded.is_admin === "True") {
          notify.show(response.data.message, "success", 2000);
          this.props.history.push("/admin");
          // customer
        } else {
          notify.show(response.data.message, "success", 2000);
          this.props.history.push("/user");
        }
      })
      .catch(error => {
        if (error.response) {
          notify.show(error.response.data.message, "error", 2500);
        } else if (error.request) {
          notify.show("Network error", "error", 2500);
        }
      });
  };

  onChange = event =>
    this.setState({ [event.target.name]: event.target.value });

  render() {
    const { email, password } = this.state;
    // console.log(this.props.context);
    return (
      <div className="row justify-content-md-center">
        <div className="col-md-5">
          <div className="jumbotron">
            <form onSubmit={this.handleLogin}>
              <h2 className="text-center form-signin-heading">
                Log in to your account
              </h2>
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input
                  type="email"
                  value={email}
                  onChange={this.onChange}
                  name="email"
                  required
                  className="form-control"
                />
              </div>
              <div className="form-group">
                <label htmlFor="password">Password</label>
                <input
                  type="password"
                  value={password}
                  onChange={this.onChange}
                  name="password"
                  required
                  className="form-control"
                />
              </div>
              <div>
                <button type="submit" className="btn btn-primary btn-block">
                  Login
                </button>
              </div>
              <p />
              <p className="text-center">
                Don't have an account yet?
                <Link to="/signup"> Sign Up</Link>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default React.forwardRef((props, ref) => (
  <AppContext.Consumer>
    {context => <Login {...props} context={context} ref={ref} />}
  </AppContext.Consumer>
));
