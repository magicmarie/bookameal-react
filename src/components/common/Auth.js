// AuthRoute: to protect authenticated user routes: user
import React from "react";
import { Redirect, Route } from "react-router-dom";

const AuthRoute = ({ component: Component, ...rest }) => {
  // check the localStorage for the token: if missing, redirect to login.
  // otherwise load the component
  const token = localStorage.getItem("token");
  return (
    <Route
      {...rest}
      render={props => {
        return token === null ? (
          <Redirect to="/login" />
        ) : (
          <Component {...props} />
        );
      }}
    />
  );
};

export default AuthRoute;
