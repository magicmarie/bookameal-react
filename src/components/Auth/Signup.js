import React from "react";
import { notify } from "react-notify-toast";
import { Link } from "react-router-dom";
import axiosInstance from "../common/Apicalls";
import { Form } from "../common/Helper";

/**
 *@param {Event} event
 * @class Signup
 * @extends {React.Component}
 */
class Signup extends React.Component {
  state = {
    name: "",
    email: "",
    password: "",
    is_admin: false,
    errors: {}
  };

  onChange = event => {
    // if name is is_admin, do this
    if (event.target.name === "is_admin") {
      this.setState({
        is_admin: event.target.checked
      });
    } else {
      this.setState({
        [event.target.name]: event.target.value
      });
    }
  };

  // sign up
  handleSignup = event => {
    event.preventDefault();
    const { name, email, password, is_admin, errors } = this.state;
    axiosInstance
      .post("/auth/signup", { name, email, password, is_admin, errors })
      .then(response => {
        notify.show(response.data.message, "success", 2500);
        this.props.history.push("/login");
      })
      .catch(err => {
        if (err.response) {
          notify.show(err.response.data.message, "error", 2500);
        }
      });
  };

  /**
   * @returns {any} rendered items
   * @memberof Signup
   */
  render() {
    const { email, name, password, is_admin } = this.state;
    const formFields = [
      {
        type: "text",
        name: "name",
        value: name,
        label: "Name"
      },
      {
        type: "email",
        name: "email",
        value: email,
        label: "Email"
      },
      {
        type: "password",
        name: "password",
        value: password,
        label: "Password"
      }
    ];
    return (
      <div className="row justify-content-md-center">
        <div className="col-md-5">
          <div className="jumbotron">
            <form onSubmit={this.handleSignup}>
              <h3 className=" text-center form-signin-heading">
                New user, Create an account
              </h3>
              <Form formFields={formFields} onChange={this.onChange} />
              <div className="form-check form-group">
                <input
                  type="checkbox"
                  name="is_admin"
                  checked={is_admin}
                  onChange={this.onChange}
                  className="form-check-input"
                />
                <label className="form-check-label" htmlFor="isadmin">
                  Caterer
                </label>
                <p className="danger">For caterer only</p>
              </div>
              <div>
                <button type="submit" className="btn btn-primary btn-block">
                  Signup
                </button>
              </div>
              <p />
              <p className="text-center">
                Already have an account?
                <Link href="/login" to="/login">
                  {" "}
                  Login
                </Link>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default Signup;
