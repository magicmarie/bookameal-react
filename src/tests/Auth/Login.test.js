import React from "react";
import { shallow } from "enzyme";
import MockAdapter from "axios-mock-adapter";
import { Login } from "../../components/Auth/Login";
import axiosInstance from "../../components/common/Apicalls";

describe("login", () => {
  let wrapper;
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn();
    const context = {
      login: mockFn
    };

    const history = {
      push: jest.fn()
    };

    wrapper = shallow(<Login context={context} history={history} />);
  });

  it("change state", () => {
    const event = {
      target: {
        name: "email",
        value: "marie@gmail.com"
      }
    };
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state.email).toEqual(event.target.value);
  });

  it("should handle login", async () => {
    const axiosMock = new MockAdapter(axiosInstance);
    axiosMock.onPost("/auth/login").reply(201, {
      message: "User created successfully",
      token: localStorage.getItem("token")
    });
    const evt = {
      preventDefault: mockFn
    };
    await wrapper.instance().handleLogin(evt);
    expect(mockFn.mock.calls.length).toEqual(1);
  });

  it("should successfully handle errors on login", () => {
    const axiosMock = new MockAdapter(axiosInstance);
    axiosMock.onPost("/auth/login").reply(400, {
      message: "failed",
      token: localStorage.getItem("token")
    });
    const evt = {
      preventDefault: mockFn
    };
    wrapper.instance().handleLogin(evt);
    expect(mockFn.mock.calls.length).toEqual(1);
  });

  it("should render successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
