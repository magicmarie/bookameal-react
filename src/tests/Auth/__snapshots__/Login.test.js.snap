// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should render successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    context={
        Object {
            "login": [Function],
          }
    }
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-md-5"
>
        <div
                className="jumbotron"
        >
                <form
                        onSubmit={[Function]}
                >
                        <h2
                                className="text-center form-signin-heading"
                        >
                                Log in to your account
                        </h2>
                        <div
                                className="form-group"
                        >
                                <label
                                        htmlFor="email"
                                >
                                        Email
                                </label>
                                <input
                                        className="form-control"
                                        name="email"
                                        onChange={[Function]}
                                        required={true}
                                        type="email"
                                        value=""
                                />
                        </div>
                        <div
                                className="form-group"
                        >
                                <label
                                        htmlFor="password"
                                >
                                        Password
                                </label>
                                <input
                                        className="form-control"
                                        name="password"
                                        onChange={[Function]}
                                        required={true}
                                        type="password"
                                        value=""
                                />
                        </div>
                        <div>
                                <button
                                        className="btn btn-primary btn-block"
                                        type="submit"
                                >
                                        Login
                                </button>
                        </div>
                        <p />
                        <p
                                className="text-center"
                        >
                                Do not have an account yet?
                                <Link
                                        href="/signup"
                                        replace={false}
                                        to="/signup"
                                >
                                         
                                        Sign Up
                                </Link>
                        </p>
                </form>
        </div>
</div>,
      "className": "row justify-content-md-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="jumbotron"
>
          <form
                    onSubmit={[Function]}
          >
                    <h2
                              className="text-center form-signin-heading"
                    >
                              Log in to your account
                    </h2>
                    <div
                              className="form-group"
                    >
                              <label
                                        htmlFor="email"
                              >
                                        Email
                              </label>
                              <input
                                        className="form-control"
                                        name="email"
                                        onChange={[Function]}
                                        required={true}
                                        type="email"
                                        value=""
                              />
                    </div>
                    <div
                              className="form-group"
                    >
                              <label
                                        htmlFor="password"
                              >
                                        Password
                              </label>
                              <input
                                        className="form-control"
                                        name="password"
                                        onChange={[Function]}
                                        required={true}
                                        type="password"
                                        value=""
                              />
                    </div>
                    <div>
                              <button
                                        className="btn btn-primary btn-block"
                                        type="submit"
                              >
                                        Login
                              </button>
                    </div>
                    <p />
                    <p
                              className="text-center"
                    >
                              Do not have an account yet?
                              <Link
                                        href="/signup"
                                        replace={false}
                                        to="/signup"
                              >
                                         
                                        Sign Up
                              </Link>
                    </p>
          </form>
</div>,
        "className": "col-md-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            onSubmit={[Function]}
>
            <h2
                        className="text-center form-signin-heading"
            >
                        Log in to your account
            </h2>
            <div
                        className="form-group"
            >
                        <label
                                    htmlFor="email"
                        >
                                    Email
                        </label>
                        <input
                                    className="form-control"
                                    name="email"
                                    onChange={[Function]}
                                    required={true}
                                    type="email"
                                    value=""
                        />
            </div>
            <div
                        className="form-group"
            >
                        <label
                                    htmlFor="password"
                        >
                                    Password
                        </label>
                        <input
                                    className="form-control"
                                    name="password"
                                    onChange={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                        />
            </div>
            <div>
                        <button
                                    className="btn btn-primary btn-block"
                                    type="submit"
                        >
                                    Login
                        </button>
            </div>
            <p />
            <p
                        className="text-center"
            >
                        Do not have an account yet?
                        <Link
                                    href="/signup"
                                    replace={false}
                                    to="/signup"
                        >
                                     
                                    Sign Up
                        </Link>
            </p>
</form>,
          "className": "jumbotron",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="text-center form-signin-heading"
>
                Log in to your account
</h2>,
              <div
                className="form-group"
>
                <label
                                htmlFor="email"
                >
                                Email
                </label>
                <input
                                className="form-control"
                                name="email"
                                onChange={[Function]}
                                required={true}
                                type="email"
                                value=""
                />
</div>,
              <div
                className="form-group"
>
                <label
                                htmlFor="password"
                >
                                Password
                </label>
                <input
                                className="form-control"
                                name="password"
                                onChange={[Function]}
                                required={true}
                                type="password"
                                value=""
                />
</div>,
              <div>
                <button
                                className="btn btn-primary btn-block"
                                type="submit"
                >
                                Login
                </button>
</div>,
              <p />,
              <p
                className="text-center"
>
                Do not have an account yet?
                <Link
                                href="/signup"
                                replace={false}
                                to="/signup"
                >
                                 
                                Sign Up
                </Link>
</p>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log in to your account",
                "className": "text-center form-signin-heading",
              },
              "ref": null,
              "rendered": "Log in to your account",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="email"
>
                    Email
</label>,
                  <input
                    className="form-control"
                    name="email"
                    onChange={[Function]}
                    required={true}
                    type="email"
                    value=""
/>,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email",
                    "htmlFor": "email",
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "email",
                    "onChange": [Function],
                    "required": true,
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    htmlFor="password"
>
                    Password
</label>,
                  <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    required={true}
                    type="password"
                    value=""
/>,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                    "htmlFor": "password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "password",
                    "onChange": [Function],
                    "required": true,
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary btn-block"
                  type="submit"
>
                  Login
</button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "btn btn-primary btn-block",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Do not have an account yet?",
                  <Link
                    href="/signup"
                    replace={false}
                    to="/signup"
>
                     
                    Sign Up
</Link>,
                ],
                "className": "text-center",
              },
              "ref": null,
              "rendered": Array [
                "Do not have an account yet?",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      " ",
                      "Sign Up",
                    ],
                    "href": "/signup",
                    "replace": false,
                    "to": "/signup",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "Sign Up",
                  ],
                  "type": [Function],
                },
              ],
              "type": "p",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-md-5"
>
          <div
                    className="jumbotron"
          >
                    <form
                              onSubmit={[Function]}
                    >
                              <h2
                                        className="text-center form-signin-heading"
                              >
                                        Log in to your account
                              </h2>
                              <div
                                        className="form-group"
                              >
                                        <label
                                                  htmlFor="email"
                                        >
                                                  Email
                                        </label>
                                        <input
                                                  className="form-control"
                                                  name="email"
                                                  onChange={[Function]}
                                                  required={true}
                                                  type="email"
                                                  value=""
                                        />
                              </div>
                              <div
                                        className="form-group"
                              >
                                        <label
                                                  htmlFor="password"
                                        >
                                                  Password
                                        </label>
                                        <input
                                                  className="form-control"
                                                  name="password"
                                                  onChange={[Function]}
                                                  required={true}
                                                  type="password"
                                                  value=""
                                        />
                              </div>
                              <div>
                                        <button
                                                  className="btn btn-primary btn-block"
                                                  type="submit"
                                        >
                                                  Login
                                        </button>
                              </div>
                              <p />
                              <p
                                        className="text-center"
                              >
                                        Do not have an account yet?
                                        <Link
                                                  href="/signup"
                                                  replace={false}
                                                  to="/signup"
                                        >
                                                   
                                                  Sign Up
                                        </Link>
                              </p>
                    </form>
          </div>
</div>,
        "className": "row justify-content-md-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="jumbotron"
>
            <form
                        onSubmit={[Function]}
            >
                        <h2
                                    className="text-center form-signin-heading"
                        >
                                    Log in to your account
                        </h2>
                        <div
                                    className="form-group"
                        >
                                    <label
                                                htmlFor="email"
                                    >
                                                Email
                                    </label>
                                    <input
                                                className="form-control"
                                                name="email"
                                                onChange={[Function]}
                                                required={true}
                                                type="email"
                                                value=""
                                    />
                        </div>
                        <div
                                    className="form-group"
                        >
                                    <label
                                                htmlFor="password"
                                    >
                                                Password
                                    </label>
                                    <input
                                                className="form-control"
                                                name="password"
                                                onChange={[Function]}
                                                required={true}
                                                type="password"
                                                value=""
                                    />
                        </div>
                        <div>
                                    <button
                                                className="btn btn-primary btn-block"
                                                type="submit"
                                    >
                                                Login
                                    </button>
                        </div>
                        <p />
                        <p
                                    className="text-center"
                        >
                                    Do not have an account yet?
                                    <Link
                                                href="/signup"
                                                replace={false}
                                                to="/signup"
                                    >
                                                 
                                                Sign Up
                                    </Link>
                        </p>
            </form>
</div>,
          "className": "col-md-5",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              onSubmit={[Function]}
>
              <h2
                            className="text-center form-signin-heading"
              >
                            Log in to your account
              </h2>
              <div
                            className="form-group"
              >
                            <label
                                          htmlFor="email"
                            >
                                          Email
                            </label>
                            <input
                                          className="form-control"
                                          name="email"
                                          onChange={[Function]}
                                          required={true}
                                          type="email"
                                          value=""
                            />
              </div>
              <div
                            className="form-group"
              >
                            <label
                                          htmlFor="password"
                            >
                                          Password
                            </label>
                            <input
                                          className="form-control"
                                          name="password"
                                          onChange={[Function]}
                                          required={true}
                                          type="password"
                                          value=""
                            />
              </div>
              <div>
                            <button
                                          className="btn btn-primary btn-block"
                                          type="submit"
                            >
                                          Login
                            </button>
              </div>
              <p />
              <p
                            className="text-center"
              >
                            Do not have an account yet?
                            <Link
                                          href="/signup"
                                          replace={false}
                                          to="/signup"
                            >
                                           
                                          Sign Up
                            </Link>
              </p>
</form>,
            "className": "jumbotron",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="text-center form-signin-heading"
>
                  Log in to your account
</h2>,
                <div
                  className="form-group"
>
                  <label
                                    htmlFor="email"
                  >
                                    Email
                  </label>
                  <input
                                    className="form-control"
                                    name="email"
                                    onChange={[Function]}
                                    required={true}
                                    type="email"
                                    value=""
                  />
</div>,
                <div
                  className="form-group"
>
                  <label
                                    htmlFor="password"
                  >
                                    Password
                  </label>
                  <input
                                    className="form-control"
                                    name="password"
                                    onChange={[Function]}
                                    required={true}
                                    type="password"
                                    value=""
                  />
</div>,
                <div>
                  <button
                                    className="btn btn-primary btn-block"
                                    type="submit"
                  >
                                    Login
                  </button>
</div>,
                <p />,
                <p
                  className="text-center"
>
                  Do not have an account yet?
                  <Link
                                    href="/signup"
                                    replace={false}
                                    to="/signup"
                  >
                                     
                                    Sign Up
                  </Link>
</p>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log in to your account",
                  "className": "text-center form-signin-heading",
                },
                "ref": null,
                "rendered": "Log in to your account",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="email"
>
                      Email
</label>,
                    <input
                      className="form-control"
                      name="email"
                      onChange={[Function]}
                      required={true}
                      type="email"
                      value=""
/>,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email",
                      "htmlFor": "email",
                    },
                    "ref": null,
                    "rendered": "Email",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "email",
                      "onChange": [Function],
                      "required": true,
                      "type": "email",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label
                      htmlFor="password"
>
                      Password
</label>,
                    <input
                      className="form-control"
                      name="password"
                      onChange={[Function]}
                      required={true}
                      type="password"
                      value=""
/>,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                      "htmlFor": "password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "password",
                      "onChange": [Function],
                      "required": true,
                      "type": "password",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="btn btn-primary btn-block"
                    type="submit"
>
                    Login
</button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                    "className": "btn btn-primary btn-block",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Do not have an account yet?",
                    <Link
                      href="/signup"
                      replace={false}
                      to="/signup"
>
                       
                      Sign Up
</Link>,
                  ],
                  "className": "text-center",
                },
                "ref": null,
                "rendered": Array [
                  "Do not have an account yet?",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        " ",
                        "Sign Up",
                      ],
                      "href": "/signup",
                      "replace": false,
                      "to": "/signup",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "Sign Up",
                    ],
                    "type": [Function],
                  },
                ],
                "type": "p",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
